fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java

```diff
@@ -745,31 +745,6 @@ public void mutateMetaData(@Nonnull Consumer<RecordMetaDataProto.MetaData.Builde
         context.asyncToSync(FDBStoreTimer.Waits.WAIT_MUTATE_METADATA, mutateMetaDataAsync(mutateMetaDataProto));
     }
 
-    /**
-     * Mutate the stored meta-data proto and record meta-data builder using mutation callbacks.
-     *
-     * <p>
-     * This method applies the given mutation callbacks on the stored meta-data and then saves it back to the store. Note the
-     * order that the callbacks are executed. {@code mutateMetaDataProto} is called first on the meta-data proto and then the
-     * second {@code mutateRecordMetaDataBuilder} is executed on the meta-data builder.
-     * Callers might need to provide an appropriate {@link MetaDataEvolutionValidator} to the meta-data store, depending on the
-     * changes that the callbacks perform.
-     * </p>
-     *
-     * <p>
-     * Also, callers must be cautious about modifying the stored records descriptor using mutation callbacks, as this will make the
-     * meta-data store the ultimate (and sole) source of truth for the record definitions and not just the in-database
-     * copy of a {@code .proto} file under source control.
-     * </p>
-     *
-     * @param mutateMetaDataProto a callback that mutates the meta-data proto
-     * @param mutateRecordMetaDataBuilder a callback that mutates the record meta-data builder after the meta-data proto is mutated
-     */
-    public void mutateMetaData(@Nonnull Consumer<RecordMetaDataProto.MetaData.Builder> mutateMetaDataProto,
-                               @Nullable Consumer<RecordMetaDataBuilder> mutateRecordMetaDataBuilder) {
-        context.asyncToSync(FDBStoreTimer.Waits.WAIT_MUTATE_METADATA, mutateMetaDataAsync(mutateMetaDataProto, mutateRecordMetaDataBuilder));
-    }
```

Comment:
I think these methods were still valuable, even if they're no longer as necessary given that the primary key is set through the `addRecordType` API. In particular, I could see someone wanting to add a new record type and then define a few new indexes on that type all in one go.

Reply:
Yeah, I thought about keeping them. But they unnecessarily complicate the the method. We already have ways to add index in `FDBMetaDataStore`. Probably that's enough. Thanks for the detailed review.

Reply:
I see. Maybe that's fine for now, but I'm still somewhat concerned that there isn't an easy way in the API right now to atomically add a new record type with a few indexes on that type that doesn't involve hand-crafting the meta-data proto.

Reply:
that's right, we cannot do it all in one place. Then, I'll revive it back, and probably add some tests to exercise that path too.

Reply:
The callback is back.